### PR TITLE
Make all TokenService validators not depend on `ConfigProvider`

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Make all TokenService validators not depend on `ConfigProvider`

### CODE REVIEW

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/StakingValidator.java [hunk 1]
```diff
@@ -37,31 +37,29 @@
...
+     * @param accountStore           readable account store
+     * @param context
      */
     public void validateStakedId(
             @NonNull final boolean hasDeclineRewardChange,
             @NonNull final String stakedIdKind,
             @Nullable final AccountID stakedAccountIdInOp,
             @Nullable final Long stakedNodeIdInOp,
-            @NonNull ReadableAccountStore accountStore) {
+            @NonNull ReadableAccountStore accountStore,
```
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/StakingValidator.java:58]: nit: missing a final here

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenSupplyChangeOpsValidator.java [hunk 2]
```diff
@@ -136,9 +136,10 @@ public void validateWipe(final long fungibleCount, final List<Long> nftSerialNum
      * @param batchSizeGetter The function to get the corresponding batch size for the token operation.
      */
     private void validateCommon(
-            final long fungibleCount, final int nftCount, @NonNull final ToIntFunction<TokensConfig> batchSizeGetter) {
-        final var tokensConfig = configProvider.getConfiguration().getConfigData(TokensConfig.class);
-
+            final long fungibleCount,
+            final int nftCount,
+            @NonNull final ToIntFunction<TokensConfig> batchSizeGetter,
+            final TokensConfig tokensConfig) {
```
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenSupplyChangeOpsValidator.java:142]: Nit: missing `@NonNull`

**Review by mhess-swl — APPROVED** [review]
A couple nits but not enough to keep from approval. Well done! :)

**Review by rbair23 — APPROVED** [review]

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java [hunk 3]
```diff
@@ -41,5 +42,20 @@ public interface NodeInfo {
...
      */
+    @NonNull
     AccountID accountOf(long nodeId);
+
+    /**
+     * Returns if the given node id is valid and the address book contains the id.
+     * @param nodeId the id of interest
+     * @return true if the given node id is valid. False otherwise.
+     */
+    default boolean isValidId(long nodeId) {
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java:53]: Will do in next PR. Thanks Matt

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenSupplyChangeOpsValidator.java (hunk 2 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenSupplyChangeOpsValidator.java:142]: Will do in next PR. Thanks Matt

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/StakingValidator.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/StakingValidator.java:58]: Will do in next PR. Thanks Matt

**Pull request merged**